lib/python/qmk/constants.py

```diff
@@ -7,3 +7,13 @@
 
 # This is the number of directories under `qmk_firmware/keyboards` that will be traversed. This is currently a limitation of our make system.
 MAX_KEYBOARD_SUBFOLDERS = 5
+
+# These mirror C constants with the same name
+DYNAMIC_KEYMAP_EEPROM_MAX_ADDR = 1023
```

Comment:
This isn't always true.  But ... I'm not sure that this is the place to change it.  

For AVR, this ... isn't always true, as well.   Maybe we need a way to define the EEPROM size better?   Especially if you're using external EEPROM.

Reply:
This is true in the sense that the QMK code also uses this value.

Though TBH while I like the idea of it automatically reducing the number of layers being used, that could also be done manually if the compile errors out due to the default number of layers not fitting the EEPROM size, i.e. it might be easier to just drop the calculation than worry about keeping this value and others like EECONFIG_SIZE in sync with QMK code.

Reply:
Generally the target audience of this command might not fully understand the implications of the error message that gets produced, nor what setting they should change to get a working firmware.
```
 #    error Dynamic keymaps are configured to use more EEPROM than is available
```

Maybe theres a better overall solution for this issue?

Reply:
I was just saying that four layers not fitting in EEPROM happens infrequently (matrix sizes greater than TKL) and there are multiple ways of resolving it, reducing layers is one, increasing EEPROM usage (if available) is another. We could add more detail to the #error and direct it to VIA documentation for ways of resolving the problem, rather than spend too much effort automagically fixing it in this script and adding more dependencies on QMK constants... 

(This is an admission that the EEPROM size constant being used now in the VIA code in QMK is not that great, I didn't want to tie it into using AVR constants and the STM EEPROM is emulated and not even a fixed constant).

Reply:
Thanks for all the feedback!
Tbh, I hate duplicating QMK constants here. (Maybe reading them from the C files?)

My vision for this command was to reduce the manual labor, produce a "conservative" keymap that will work on the board and leave fine-tuning to the porter.

Reply:
@Erovia Agreed. Perhaps all you need is a basic calculation, ensure the keymap layers fit in 768 bytes, the remaining 256 or more bytes will be used by QMK settings, VIA settings, and macros.

Reply:
My concern here, is that now that we have the external eeprom driver, we'll see more and more boards using it, which ... will muddy this even more.   16+KB of EEPROM is ... nice to have

Reply:
How is that relevant to this script which as @Erovia stated, is just meant to produce a "conservative" setup for VIA and leave fine-tuning such as using more than 1K up to the user?